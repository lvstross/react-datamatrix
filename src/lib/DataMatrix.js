import { jsx as _jsx } from "react/jsx-runtime";
import DATAMatrix from './utils/datamatrix';
export default function DataMatrix({ msg = '', dim = 256, rct = 0, pad = 2, pal = ["#000000", "#f2f4f8"], vrb = 0, imgAttrs }) {
    const dm = DATAMatrix({ msg, dim, rct, pad, pal, vrb });
    return (_jsx("img", { ...imgAttrs, src: `data:image/svg+xml;base64,${window.btoa(dm?.outerHTML)}` }));
}
