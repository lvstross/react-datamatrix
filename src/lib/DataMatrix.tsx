/**
	https://github.com/lvstross/react-datamatrix
	License: MIT
	Copyright (c) 2022 Levi Gonzales
*/

import React from 'react'
import DATAMatrix, { DataMatrixArgs } from './utils/datamatrix'

export interface DataMatrixProps extends DataMatrixArgs {
  imgAttrs?: React.ImgHTMLAttributes<HTMLImageElement>;
}

export default function DataMatrix({
  msg = '',
  dim = 256,
  rct = 0,
  pad = 2,
  pal = ["#000000", "#f2f4f8"],
  vrb = 0,
  imgAttrs
}: DataMatrixProps) {
  const dm = DATAMatrix({ msg, dim, rct, pad, pal, vrb }) as HTMLElement

  return (
    <img
      {...imgAttrs}
      src={`data:image/svg+xml;base64,${window.btoa(dm?.outerHTML)}`}
    />
  )
}

