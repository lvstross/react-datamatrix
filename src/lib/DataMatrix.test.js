import { jsx as _jsx } from "react/jsx-runtime";
import renderer from 'react-test-renderer';
import { describe, expect, test } from 'vitest';
import DataMatrix from './DataMatrix';
describe('DataMatrix', () => {
    test('DataMatrix component renders correctly', () => {
        const component = renderer.create(_jsx(DataMatrix, { msg: "hello world" }));
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('DataMatrix component renders with props correctly', () => {
        const props = {
            msg: 'Hello World',
            dim: 300,
            rct: 1,
            pad: 5,
            pal: ["#000", "#fff"],
            vrb: 1
        };
        const component = renderer.create(_jsx(DataMatrix, { ...props }));
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('DataMatrix component renders with img attributes', () => {
        const component = renderer.create(_jsx(DataMatrix, { msg: "hello world", imgAttrs: { alt: 'Hello World', width: "500", height: "600" } }));
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
