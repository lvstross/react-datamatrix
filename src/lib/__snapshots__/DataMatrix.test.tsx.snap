// Vitest Snapshot v1

exports[`DataMatrix > DataMatrix component renders correctly 1`] = `
<img
  src="data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjAgMjAiIHdpZHRoPSIyNTYiIGhlaWdodD0iMjU2IiBmaWxsPSIjMDAwMDAwIiBzaGFwZS1yZW5kZXJpbmc9ImNyaXNwRWRnZXMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIj48cGF0aCBmaWxsPSIjZjJmNGY4IiBkPSJNMCwwdjIwaDIwVjBIMFoiPjwvcGF0aD48cGF0aCB0cmFuc2Zvcm09Im1hdHJpeCgxLDAsMCwxLDIsMikiIGQ9Ik0wLDE1aDE2djFoLTE2di0xek0xNCwxNGgxdjFoLTF2LTF6TTExLDE0aDJ2MWgtMnYtMXpNNywxNGgydjFoLTJ2LTF6TTUsMTRoMXYxaC0xdi0xek0yLDE0aDF2MWgtMXYtMXpNMCwxNGgxdjFoLTF2LTF6TTE1LDEzaDF2MWgtMXYtMXpNMTIsMTNoMnYxaC0ydi0xek04LDEzaDN2MWgtM3YtMXpNMywxM2gxdjFoLTF2LTF6TTAsMTNoMXYxaC0xdi0xek0xMiwxMmgydjFoLTJ2LTF6TTksMTJoMnYxaC0ydi0xek02LDEyaDJ2MWgtMnYtMXpNMCwxMmg1djFoLTV2LTF6TTE1LDExaDF2MWgtMXYtMXpNMTEsMTFoM3YxaC0zdi0xek04LDExaDJ2MWgtMnYtMXpNNiwxMWgxdjFoLTF2LTF6TTMsMTFoMnYxaC0ydi0xek0wLDExaDF2MWgtMXYtMXpNMywxMGgxdjFoLTF2LTF6TTAsMTBoMnYxaC0ydi0xek0xNSw5aDF2MWgtMXYtMXpNMTMsOWgxdjFoLTF2LTF6TTgsOWgxdjFoLTF2LTF6TTYsOWgxdjFoLTF2LTF6TTAsOWg1djFoLTV2LTF6TTExLDhoNHYxaC00di0xek0zLDhoNHYxaC00di0xek0wLDhoMXYxaC0xdi0xek0xMyw3aDN2MWgtM3YtMXpNMTAsN2gydjFoLTJ2LTF6TTcsN2gxdjFoLTF2LTF6TTMsN2gydjFoLTJ2LTF6TTAsN2gxdjFoLTF2LTF6TTEzLDZoMXYxaC0xdi0xek05LDZoM3YxaC0zdi0xek00LDZoMXYxaC0xdi0xek0yLDZoMXYxaC0xdi0xek0wLDZoMXYxaC0xdi0xek0xMyw1aDN2MWgtM3YtMXpNOCw1aDJ2MWgtMnYtMXpNNSw1aDF2MWgtMXYtMXpNMCw1aDN2MWgtM3YtMXpNMTMsNGgxdjFoLTF2LTF6TTEwLDRoMnYxaC0ydi0xek04LDRoMXYxaC0xdi0xek0wLDRoNXYxaC01di0xek0xMywzaDN2MWgtM3YtMXpNMTAsM2gydjFoLTJ2LTF6TTcsM2gxdjFoLTF2LTF6TTQsM2gxdjFoLTF2LTF6TTAsM2gzdjFoLTN2LTF6TTEyLDJoMnYxaC0ydi0xek0xMCwyaDF2MWgtMXYtMXpNNCwyaDF2MWgtMXYtMXpNMCwyaDF2MWgtMXYtMXpNMTAsMWg2djFoLTZ2LTF6TTYsMWgxdjFoLTF2LTF6TTQsMWgxdjFoLTF2LTF6TTAsMWgydjFoLTJ2LTF6TTE0LDBoMXYxaC0xdi0xek0xMiwwaDF2MWgtMXYtMXpNMTAsMGgxdjFoLTF2LTF6TTgsMGgxdjFoLTF2LTF6TTYsMGgxdjFoLTF2LTF6TTQsMGgxdjFoLTF2LTF6TTIsMGgxdjFoLTF2LTF6TTAsMGgxdjFoLTF2LTF6Ij48L3BhdGg+PC9zdmc+"
/>
`;

exports[`DataMatrix > DataMatrix component renders with img attributes 1`] = `
<img
  alt="Hello World"
  height="600"
  src="data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMjAgMjAiIHdpZHRoPSIyNTYiIGhlaWdodD0iMjU2IiBmaWxsPSIjMDAwMDAwIiBzaGFwZS1yZW5kZXJpbmc9ImNyaXNwRWRnZXMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIj48cGF0aCBmaWxsPSIjZjJmNGY4IiBkPSJNMCwwdjIwaDIwVjBIMFoiPjwvcGF0aD48cGF0aCB0cmFuc2Zvcm09Im1hdHJpeCgxLDAsMCwxLDIsMikiIGQ9Ik0wLDE1aDE2djFoLTE2di0xek0xNCwxNGgxdjFoLTF2LTF6TTExLDE0aDJ2MWgtMnYtMXpNNywxNGgydjFoLTJ2LTF6TTUsMTRoMXYxaC0xdi0xek0yLDE0aDF2MWgtMXYtMXpNMCwxNGgxdjFoLTF2LTF6TTE1LDEzaDF2MWgtMXYtMXpNMTIsMTNoMnYxaC0ydi0xek04LDEzaDN2MWgtM3YtMXpNMywxM2gxdjFoLTF2LTF6TTAsMTNoMXYxaC0xdi0xek0xMiwxMmgydjFoLTJ2LTF6TTksMTJoMnYxaC0ydi0xek02LDEyaDJ2MWgtMnYtMXpNMCwxMmg1djFoLTV2LTF6TTE1LDExaDF2MWgtMXYtMXpNMTEsMTFoM3YxaC0zdi0xek04LDExaDJ2MWgtMnYtMXpNNiwxMWgxdjFoLTF2LTF6TTMsMTFoMnYxaC0ydi0xek0wLDExaDF2MWgtMXYtMXpNMywxMGgxdjFoLTF2LTF6TTAsMTBoMnYxaC0ydi0xek0xNSw5aDF2MWgtMXYtMXpNMTMsOWgxdjFoLTF2LTF6TTgsOWgxdjFoLTF2LTF6TTYsOWgxdjFoLTF2LTF6TTAsOWg1djFoLTV2LTF6TTExLDhoNHYxaC00di0xek0zLDhoNHYxaC00di0xek0wLDhoMXYxaC0xdi0xek0xMyw3aDN2MWgtM3YtMXpNMTAsN2gydjFoLTJ2LTF6TTcsN2gxdjFoLTF2LTF6TTMsN2gydjFoLTJ2LTF6TTAsN2gxdjFoLTF2LTF6TTEzLDZoMXYxaC0xdi0xek05LDZoM3YxaC0zdi0xek00LDZoMXYxaC0xdi0xek0yLDZoMXYxaC0xdi0xek0wLDZoMXYxaC0xdi0xek0xMyw1aDN2MWgtM3YtMXpNOCw1aDJ2MWgtMnYtMXpNNSw1aDF2MWgtMXYtMXpNMCw1aDN2MWgtM3YtMXpNMTMsNGgxdjFoLTF2LTF6TTEwLDRoMnYxaC0ydi0xek04LDRoMXYxaC0xdi0xek0wLDRoNXYxaC01di0xek0xMywzaDN2MWgtM3YtMXpNMTAsM2gydjFoLTJ2LTF6TTcsM2gxdjFoLTF2LTF6TTQsM2gxdjFoLTF2LTF6TTAsM2gzdjFoLTN2LTF6TTEyLDJoMnYxaC0ydi0xek0xMCwyaDF2MWgtMXYtMXpNNCwyaDF2MWgtMXYtMXpNMCwyaDF2MWgtMXYtMXpNMTAsMWg2djFoLTZ2LTF6TTYsMWgxdjFoLTF2LTF6TTQsMWgxdjFoLTF2LTF6TTAsMWgydjFoLTJ2LTF6TTE0LDBoMXYxaC0xdi0xek0xMiwwaDF2MWgtMXYtMXpNMTAsMGgxdjFoLTF2LTF6TTgsMGgxdjFoLTF2LTF6TTYsMGgxdjFoLTF2LTF6TTQsMGgxdjFoLTF2LTF6TTIsMGgxdjFoLTF2LTF6TTAsMGgxdjFoLTF2LTF6Ij48L3BhdGg+PC9zdmc+"
  width="500"
/>
`;

exports[`DataMatrix > DataMatrix component renders with props correctly 1`] = `
<img
  src="data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgMzYgMjIiIHdpZHRoPSI0OTAiIGhlaWdodD0iMzAwIiBmaWxsPSIjMDAwIiBzaGFwZS1yZW5kZXJpbmc9ImNyaXNwRWRnZXMiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIj48cGF0aCBmaWxsPSIjZmZmIiBkPSJNMCwwdjIyaDM2VjBIMFoiPjwvcGF0aD48cGF0aCB0cmFuc2Zvcm09Im1hdHJpeCgxLDAsMCwxLDUsNSkiIGQ9Ik0yNSwxMWgxdjFoLTF2LTF6TTI0LDExaDF2MWgtMXYtMXpNMjMsMTFoMXYxaC0xdi0xek0yMiwxMWgxdjFoLTF2LTF6TTIxLDExaDF2MWgtMXYtMXpNMjAsMTFoMXYxaC0xdi0xek0xOSwxMWgxdjFoLTF2LTF6TTE4LDExaDF2MWgtMXYtMXpNMTcsMTFoMXYxaC0xdi0xek0xNiwxMWgxdjFoLTF2LTF6TTE1LDExaDF2MWgtMXYtMXpNMTQsMTFoMXYxaC0xdi0xek0xMywxMWgxdjFoLTF2LTF6TTEyLDExaDF2MWgtMXYtMXpNMTEsMTFoMXYxaC0xdi0xek0xMCwxMWgxdjFoLTF2LTF6TTksMTFoMXYxaC0xdi0xek04LDExaDF2MWgtMXYtMXpNNywxMWgxdjFoLTF2LTF6TTYsMTFoMXYxaC0xdi0xek01LDExaDF2MWgtMXYtMXpNNCwxMWgxdjFoLTF2LTF6TTMsMTFoMXYxaC0xdi0xek0yLDExaDF2MWgtMXYtMXpNMSwxMWgxdjFoLTF2LTF6TTAsMTFoMXYxaC0xdi0xek0yMiwxMGgxdjFoLTF2LTF6TTE4LDEwaDF2MWgtMXYtMXpNMTUsMTBoMXYxaC0xdi0xek0xNCwxMGgxdjFoLTF2LTF6TTgsMTBoMXYxaC0xdi0xek03LDEwaDF2MWgtMXYtMXpNNSwxMGgxdjFoLTF2LTF6TTMsMTBoMXYxaC0xdi0xek0xLDEwaDF2MWgtMXYtMXpNMCwxMGgxdjFoLTF2LTF6TTI1LDloMXYxaC0xdi0xek0yMCw5aDF2MWgtMXYtMXpNMTksOWgxdjFoLTF2LTF6TTE4LDloMXYxaC0xdi0xek0xNCw5aDF2MWgtMXYtMXpNOSw5aDF2MWgtMXYtMXpNMiw5aDF2MWgtMXYtMXpNMCw5aDF2MWgtMXYtMXpNMjMsOGgxdjFoLTF2LTF6TTE5LDhoMXYxaC0xdi0xek0xNCw4aDF2MWgtMXYtMXpNMTMsOGgxdjFoLTF2LTF6TTEwLDhoMXYxaC0xdi0xek04LDhoMXYxaC0xdi0xek00LDhoMXYxaC0xdi0xek0zLDhoMXYxaC0xdi0xek0wLDhoMXYxaC0xdi0xek0yNSw3aDF2MWgtMXYtMXpNMjAsN2gxdjFoLTF2LTF6TTE5LDdoMXYxaC0xdi0xek0xNyw3aDF2MWgtMXYtMXpNMTUsN2gxdjFoLTF2LTF6TTE0LDdoMXYxaC0xdi0xek0xMyw3aDF2MWgtMXYtMXpNMTAsN2gxdjFoLTF2LTF6TTgsN2gxdjFoLTF2LTF6TTcsN2gxdjFoLTF2LTF6TTYsN2gxdjFoLTF2LTF6TTQsN2gxdjFoLTF2LTF6TTIsN2gxdjFoLTF2LTF6TTEsN2gxdjFoLTF2LTF6TTAsN2gxdjFoLTF2LTF6TTIzLDZoMXYxaC0xdi0xek0yMiw2aDF2MWgtMXYtMXpNMjEsNmgxdjFoLTF2LTF6TTIwLDZoMXYxaC0xdi0xek0xOCw2aDF2MWgtMXYtMXpNMTcsNmgxdjFoLTF2LTF6TTE2LDZoMXYxaC0xdi0xek0xNCw2aDF2MWgtMXYtMXpNMTMsNmgxdjFoLTF2LTF6TTEyLDZoMXYxaC0xdi0xek0xMSw2aDF2MWgtMXYtMXpNMTAsNmgxdjFoLTF2LTF6TTksNmgxdjFoLTF2LTF6TTgsNmgxdjFoLTF2LTF6TTYsNmgxdjFoLTF2LTF6TTUsNmgxdjFoLTF2LTF6TTQsNmgxdjFoLTF2LTF6TTEsNmgxdjFoLTF2LTF6TTAsNmgxdjFoLTF2LTF6TTI1LDVoMXYxaC0xdi0xek0yMSw1aDF2MWgtMXYtMXpNMjAsNWgxdjFoLTF2LTF6TTExLDVoMXYxaC0xdi0xek05LDVoMXYxaC0xdi0xek03LDVoMXYxaC0xdi0xek02LDVoMXYxaC0xdi0xek0yLDVoMXYxaC0xdi0xek0xLDVoMXYxaC0xdi0xek0wLDVoMXYxaC0xdi0xek0yMiw0aDF2MWgtMXYtMXpNMjAsNGgxdjFoLTF2LTF6TTE4LDRoMXYxaC0xdi0xek0xNyw0aDF2MWgtMXYtMXpNMTUsNGgxdjFoLTF2LTF6TTExLDRoMXYxaC0xdi0xek05LDRoMXYxaC0xdi0xek03LDRoMXYxaC0xdi0xek0xLDRoMXYxaC0xdi0xek0wLDRoMXYxaC0xdi0xek0yNSwzaDF2MWgtMXYtMXpNMjQsM2gxdjFoLTF2LTF6TTIyLDNoMXYxaC0xdi0xek0yMSwzaDF2MWgtMXYtMXpNMjAsM2gxdjFoLTF2LTF6TTE1LDNoMXYxaC0xdi0xek0xMSwzaDF2MWgtMXYtMXpNOSwzaDF2MWgtMXYtMXpNOCwzaDF2MWgtMXYtMXpNNSwzaDF2MWgtMXYtMXpNNCwzaDF2MWgtMXYtMXpNMiwzaDF2MWgtMXYtMXpNMSwzaDF2MWgtMXYtMXpNMCwzaDF2MWgtMXYtMXpNMjIsMmgxdjFoLTF2LTF6TTIxLDJoMXYxaC0xdi0xek0xNSwyaDF2MWgtMXYtMXpNMTMsMmgxdjFoLTF2LTF6TTEyLDJoMXYxaC0xdi0xek05LDJoMXYxaC0xdi0xek04LDJoMXYxaC0xdi0xek02LDJoMXYxaC0xdi0xek01LDJoMXYxaC0xdi0xek0xLDJoMXYxaC0xdi0xek0wLDJoMXYxaC0xdi0xek0yNSwxaDF2MWgtMXYtMXpNMjEsMWgxdjFoLTF2LTF6TTIwLDFoMXYxaC0xdi0xek0xNywxaDF2MWgtMXYtMXpNMTYsMWgxdjFoLTF2LTF6TTE0LDFoMXYxaC0xdi0xek0xMywxaDF2MWgtMXYtMXpNMTAsMWgxdjFoLTF2LTF6TTgsMWgxdjFoLTF2LTF6TTYsMWgxdjFoLTF2LTF6TTUsMWgxdjFoLTF2LTF6TTMsMWgxdjFoLTF2LTF6TTIsMWgxdjFoLTF2LTF6TTAsMWgxdjFoLTF2LTF6TTI0LDBoMXYxaC0xdi0xek0yMiwwaDF2MWgtMXYtMXpNMjAsMGgxdjFoLTF2LTF6TTE4LDBoMXYxaC0xdi0xek0xNiwwaDF2MWgtMXYtMXpNMTQsMGgxdjFoLTF2LTF6TTEyLDBoMXYxaC0xdi0xek0xMCwwaDF2MWgtMXYtMXpNOCwwaDF2MWgtMXYtMXpNNiwwaDF2MWgtMXYtMXpNNCwwaDF2MWgtMXYtMXpNMiwwaDF2MWgtMXYtMXpNMCwwaDF2MWgtMXYtMXoiPjwvcGF0aD48L3N2Zz4="
/>
`;
